website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -1994,16 +2213,23 @@ to [register a function directly](https://github.com/apache/beam/blob/master/sdk
 Note that if, as is often the case in Javascript, `func` returns objects that
 contain closures, it is not sufficient to register `func` alone--its return
 value must be registered if used.</span>
+<span class="language-rust">The traits to be implemented in user code, such
+as `DoFn`, `CombineFn`, and `WindowFn`, have `serde::Serializable` and `serde::Deserialize<'static>` constraints.</span>
```

Comment:
`serde::Serialize` and `serde::Deserialize` might be net necessary if the following idea is realized.
cc @dahlbaek 

https://github.com/laysakura/beam/pull/14
> I like the idea of avoiding serialization/deserialization of the actual closures, and a simple way seems to be to just store all the closures in a static at pipeline generation time (and of course generate the pipeline both on workers and the pipeline builder), and then only serialize/derserialize the key.